OtripleS.Web.Api/Services/StudentService.cs

```diff
@@ -26,5 +29,41 @@ public async ValueTask<Student> DeleteStudentAsync(Guid studentId)
 
             return await this.storageBroker.DeleteStudentAsync(maybeStudent);
         }
+
+        public async ValueTask<Student> ModifyStudentAsync(Guid studentId, StudentUpdateDto updateDto)
+        {
+            var student =
+                await this.storageBroker.SelectStudentByIdAsync(studentId);
+            if (student.HasValue())
+            {
+                MapChangesToStudent(updateDto, student);
+
+                return await this.storageBroker.UpdateStudentAsync(student);
+            }
+
+            return await Task.FromResult<Student>(null);
```

Comment:
`Task.FromResult` is compiler trickery, you could just return whatever storage broker gives you back, it would be much easier.
Also, if the returned value is null then we are supposed to throw `StudentNotFoundException` instead of returning a null.

Reply:
Here's an example of `ModifyStudent` function:
https://github.com/hassanhabib/SchoolEM/blob/master/SchoolEM/Services/StudentService.cs#L66